client/lib/query-args/index.ts

```diff
@@ -5,7 +5,7 @@ function getRelativeUrlWithParameters(
 	queryArgs: object,
 	clearExistingParameters = false
 ): string {
-	const url = new URL( window.location.href );
+	const url = new URL( page.current || window.location.href, window.location.origin );
```

Comment:
Can this be simplified to something like this?

```suggestion
	const url = new URL( page.current, window.location.origin );
```


Typescript tells me `page.current` is always a string, the default (`''`) is set [here](https://github.com/visionmedia/page.js/blob/master/page.js#L440). This will fall back onto the origin to create a basic url without any path.

Is there some kind of initial situation where page.current isn't set yet but there is a path in window location we should be using?

Reply:
I'm a bit unfamiliar with the code so I preferred to be extra safe and keep the previous approach as a fallback.

Reply:
I actually tried removing `window.location.href` and discovered a few things:
* `page.current` defaulting to `''` will cause and `Invalid URL` error
* I updated the unit tests and discovered there's a `getQueryArgs` which also uses `window.location.href`. Here, trying to use `page.current` is `''` and causes #64769 to not work anymore

This is making me feel apprehensive about forcing `page.current` in `setQueryArgs`.
Instead, taking a step back and using `history.pushState` (via `page.show`) and silently updating `location.href` in the search box, feels like a cleaner solution.